### DISCUSSION TITLE
The <single-file-note> would invisible when parsing with document.createRange().createContextualFragment()

### DISCUSSION THREAD

**Discussion by nuthrash** [gildas-lormeau/SingleFile#1057] [Q&A] [answered]
Sorry, it's me again,

I am coding the next version of "HTML Reader" plugin, and I encounter a strange condition.
I plan to use the `document.createRange().createContextualFragment(htmlStr)` (it would allow scripting) to parsing HTML files, this method would not support Declarative Shadow DOM(**DSD**), so I think the content of `<single-file-note>` should invisible normally.
However, the HTML file made by SingleFileZ is different, its content of `<single-file-note>` are still visible.
I wonder what is the difference between SingleFile and SingleFileZ to treat `<single-file-note>` ?

Fig.1 Star Wars - Wikipedia - SingleFile with Notes3 at Unrestricted Mode 1
![Star Wars - Wikipedia - SingleFile with Notes3 at Unrestricted Mode 1](https://user-images.githubusercontent.com/43021861/202890347-d1b28571-6517-4991-925f-44a665d4071f.jpg)

<br />

Fig.2 Star Wars - Wikipedia - SingleFileZ with Notes3 at Unrestricted Mode 1
![Star Wars - Wikipedia - SingleFileZ with Notes3 at Unrestricted Mode 1](https://user-images.githubusercontent.com/43021861/202890423-8a1391b3-9752-4498-a67c-c8bcbcebb64f.jpg)

Apparently, the `<single-file-note>` inside Fig.2 is rendered very well. I use the same method to parsing them, but got different results.

**Comment by gildas-lormeau** [comment, marked as answer]
Right now, this issue is very weird to me. Actually, by investigating on it I think I've found a bug in SingleFileZ that would prevent  the polyfill code to run. As a consequence, I would be less surprised if you inverted your screenshots, but anyway... I'll let you know very soon if the bug exists or not.

**Reply by nuthrash** [reply]
Hmm, the screenshots are right to their real situation, you can see the file name "**temp / Star Wars - Wikipedia - SingleFile with Notes3**" and "**temp / Star Wars - Wikipedia - SingleFileZ with Notes3**" on the top tabs of them. The "**Notes3**" means I re-captured the same wiki page for the 3rd time, because I found the "**Notes2**" files has same situation, I re-capture them to avoid I messed something.

Sometimes a bug is not a bug, maybe it contains a new point of view. In my opinion, I think the presentation of Fig.2 (SingleFileZ) is correct for the polyfill, instead of Fig.1 (SingleFile). Please keep the correct presentation if you can.

**Reply by gildas-lormeau** [reply]
So, I fixed the bug in both SingleFileZ and SingleFile actually. The problem is that the code creating notes was not placed after the notes in the document. Because of this, SingleFile was waiting for the `DOMContentLoaded` to be fired before executing this code. It was inconvenient and is probably the cause of the bug you described because this event was already fired. I attached 2 files saved with the fix applied. Please let me know if it changes anything.

[examples.zip](https://github.com/gildas-lormeau/SingleFile/files/10061512/examples.zip)

**Reply by gildas-lormeau** [reply]
Another way to validate it's the cause of the issue would be to execute `dispatchEvent(new CustomEvent("DOMContentLoaded"))` in your code just after you display a page saved with SingleFile.

**Reply by nuthrash** [reply]
OK, I can see the Note with "Hello from SingleFile!" text of "Example Domain (11_22_2022 1_18_16 AM).html" at "**Unrestricted Mode**" after fix its `<div>` style. Please take other website as examples next time .
This is what I expected, Excellent! :thumbsup:

I can see the Note with "Hello from SingleFileZ!" text of "Example Domain (11_22_2022 1_18_26 AM).zip.html" at normal browsers, but I cannot directly modify the `<div>` style of it for "**Unrestricted Mode**". Sorry about that, because the "**Unrestricted Mode**" does not isolate CSS styles.

> Another way to validate it's the cause of the issue would be to execute `dispatchEvent(new CustomEvent("DOMContentLoaded"))` in your code just after you display a page saved with SingleFile.

Oh, I can see the Notes of "Example Domain (11_22_2022 1_18_16 AM).html"  and previously saved files after inserted this code. I will add this patch code to the plugin. Thank you very much! :gift:
